Bugzilla/App/SES.pm

```diff
@@ -199,16 +199,32 @@ sub _process_bounce {
           mta    => $notification->{bounce}->{reportingMTA} // 'unknown',
           reason => $reason,
         };
-        my $disable_text;
+        my $bounce_message;
         $template->process('admin/users/bounce-disabled.txt.tmpl',
-          $vars, \$disable_text)
+          $vars, \$bounce_message)
           || die $template->error();
 
-        $user->set_disabledtext($disable_text);
         $user->set_disable_mail(1);
+        $user->set_bounce_count($user->bounce_count + 1);
```

Comment:
This should be _recent_ bounces, rather than a count since over the entire lifetime of an account.

As implemented if an account has 4 bounces early on, then 5 years later there's a single bounce, bugmail will be disabled.

I'm not sure what the actual timeframe should be (maybe @ckolos can provide some guidance), but something like "5 bounces within the last 4 weeks" feels about right.

Reply:
> 
> 
> This should be _recent_ bounces, rather than a count since over the entire lifetime of an account.
> 
> As implemented if an account has 4 bounces early on, then 5 years later there's a single bounce, bugmail will be disabled.
> 
> I'm not sure what the actual timeframe should be (maybe @ckolos can provide some guidance), but something like "5 bounces within the last 4 weeks" feels about right.

Actually I failed to mention here that there is also a new checkbox on the editusers.cgi user view that is for resetting the bounce count to zero. So when a user is deactivated BOUNCE_COUNT_MAX times and the admin has to enable their account manually, the admin will also need to check the box to reset the count. There is a link from the editusers.cgi view to the age where all of the bounce messages are listed similar to what the user sees.

That being said, I did toy with the idea in B/App/SES.pm of when a bounce occurs, and the user has not had a bounce in in less than 30 days, then set the bouce count to 1 instead of whatever the value currently is. I think that will work out fine for now.

Reply:
I'm still not sure having a perm bounce count is ideal, however I'm happy to leave it as-is and see how it works out.